kirin/core/merge_utils.py

```diff
@@ -360,37 +360,74 @@ def log_stu_modif(trip_update, stu, string_additional_info):
+def manage_deleted_stop_event(event_type, stu, previous_stop_event, trip_update):
+    # not considering deleted {event_type} (event_type can be departure or arrival)
+    if stu.is_stop_event_deleted(event_type):
+        return previous_stop_event
+    delay_field = "{}_delay"
+    # if {event_type} is before previous stop-event's time:
+    # push {event_type} time so that its delay is the same than for previous time
+    if previous_stop_event.time is not None and previous_stop_event.time > getattr(stu, event_type):
+        delay_diff = previous_stop_event.delay - getattr(stu, delay_field.format(event_type))
+        stu.departure_delay += delay_diff
+        setattr(stu, delay_field.format(event_type), (getattr(stu, delay_field.format(event_type)) + delay_diff))
+        setattr(stu, event_type, (getattr(stu, event_type) + delay_diff))
+        log_stu_modif(
+            trip_update,
+            stu,
+            "departure = {t} and departure_delay = {d}".format(
+                t=getattr(stu, event_type), d=getattr(stu, delay_field.format(event_type))
+            ),
+        )
+    return TimeDelayTuple(time=getattr(stu, event_type), delay=getattr(stu, delay_field.format(event_type)))
+
+
 def manage_consistency(trip_update):
     """
     receive a TripUpdate, then adjust its consistency
     """
-    logger = logging.getLogger(__name__)
+
     previous_stop_event = TimeDelayTuple(time=None, delay=None)
     for current_order, stu in enumerate(trip_update.stop_time_updates):
         # modifications
-        if stu.arrival is None:
-            stu.arrival = stu.departure
-            if stu.arrival is None and previous_stop_event.time is not None:
-                stu.arrival = previous_stop_event.time
-            if stu.arrival is None:
-                logger.warning(
-                    "TripUpdate on navitia vj {nav_id} on {date} rejected: "
-                    "StopTimeUpdate missing arrival time".format(
-                        nav_id=trip_update.vj.navitia_trip_id, date=trip_update.vj.get_circulation_date()
-                    )
-                )
-                return False
-            log_stu_modif(trip_update, stu, "arrival = {v}".format(v=stu.arrival))
-            if not stu.arrival_delay and stu.departure_delay:
-                stu.arrival_delay = stu.departure_delay
-                log_stu_modif(trip_update, stu, "arrival_delay = {v}".format(v=stu.arrival_delay))
-
-        if stu.departure is None:
-            stu.departure = stu.arrival
-            log_stu_modif(trip_update, stu, "departure = {v}".format(v=stu.departure))
-            if not stu.departure_delay and stu.arrival_delay:
-                stu.departure_delay = stu.arrival_delay
-                log_stu_modif(trip_update, stu, "departure_delay = {v}".format(v=stu.departure_delay))
+        if not manage_arrival_consistency(stu, trip_update, previous_stop_event):
+            return False
+        manage_departure_consistency(stu, trip_update)
```

Comment:
It can't happen, but I would probably do the same here (matching fn signature):
```suggestion
        if not manage_departure_consistency(stu, trip_update):
            return False
```